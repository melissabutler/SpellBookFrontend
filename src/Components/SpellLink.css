.SpellLink-link h5 {
     color: black;
     background-color: white;
     padding: 10px;
     width: 100%;
     border-radius: 10px;
     
}

