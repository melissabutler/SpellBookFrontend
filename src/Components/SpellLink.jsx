import React from "react";
import { Link } from 'react-router-dom';

import { Container } from 'react-bootstrap';

import "./SpellLink.css"

const SpellLink = ({spell, charProfile}) => {
    return (
        <Container>
           <Link className="SpellLink-link" key={spell.index} to={`/spells/${spell.index}`}>
           {charProfile === true ?
           <h5>{spell.name}</h5>
            :
            <h5>Level {spell.level}: {spell.name}</h5>
            }
            
            </Link> 
        </Container>
        
    )
}

export default SpellLink;