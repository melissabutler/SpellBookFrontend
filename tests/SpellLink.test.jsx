import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen} from '@testing-library/react'
import SpellLink from "../src/Components/SpellLink";

describe("SpellLink", function() {

    
        vi.mock('react', async (importOriginal) => {
            const react = await importOriginal('react');
            return {
                ...react,
                useContext: () => ({ currentUser: null}),
                useRef: () => { target: null }
        }})

        vi.mock('react-router-dom', async () => {
            const mod = await vi.importActual('react-router-dom');
            return {
                ...mod, 
                useNavigate: () => "/",
                BrowserRouter: () => "",
                Link: () => ""
            }
        })
    it('renders the pellLink component', async () => {
        render(
            <SpellLink spell={{index: "acid-arrow", name: "Acid Arrow"}}/>)
        screen.debug();
      
    });

        it('matches snapshot', () => {
            const { asFragment } = render(<SpellLink spell={{index: "acid-arrow", name: "Acid Arrow"}}/>);
            expect(asFragment()).toMatchSnapshot();
        });



})
   