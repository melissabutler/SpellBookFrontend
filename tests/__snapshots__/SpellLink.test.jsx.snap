// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpellLink > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  />
</DocumentFragment>
`;
